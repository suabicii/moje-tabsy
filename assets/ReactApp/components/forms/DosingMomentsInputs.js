import React from "react";
import DosingMomentInput from "./DosingMomentInput";

function DosingMomentsInputs(props) {
    const inputArray = [];

    const addInputToArray = (name, value = "") => {
        inputArray.push(<DosingMomentInput name={name} value={value}/>);
    };

    if (props.dosingMoments) {
        for (const [key, value] of props.dosingMoments) {
            addInputToArray(`hour${key}`, value);
        }
        if (props.inputAmount > props.dosingMoments.length) {
            for (let i = props.dosingMoments.length; i < props.inputAmount; i++) {
                addInputToArray(`hour${i + 1}`);
            }
        }
    } else {
        for (let i = 0; i < props.inputAmount; i++) {
            addInputToArray(`hour${i + 1}`);
        }
    }

    return (
        <>
            {inputArray.map((item, index) => (
                <div key={index}>
                    {item}
                </div>
            ))}
        </>
    );
}

export default DosingMomentsInputs;