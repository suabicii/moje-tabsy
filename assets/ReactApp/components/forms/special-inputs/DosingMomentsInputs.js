import React from "react";
import DosingMomentInput from "./DosingMomentInput";

function DosingMomentsInputs(props) {
    const inputArray = [];

    const addInputToArray = (name, value = "") => {
        inputArray.push(
            <DosingMomentInput
                handleTimeInputChange={props.handleTimeInputChange}
                name={name}
                value={value}
            />
        );
    };

    if (props.dosingMoments) {
        for (const [key, value] of props.dosingMoments) {
            if (parseInt(key.substring(key.length - 1)) > props.inputAmount) {
                break;
            }
            addInputToArray(key, value);
        }
        if (props.inputAmount > props.dosingMoments.length) {
            for (let i = props.dosingMoments.length; i < props.inputAmount; i++) {
                addInputToArray(`hour${i + 1}`);
            }
        }
    } else {
        for (let i = 0; i < props.inputAmount; i++) {
            addInputToArray(`hour${i + 1}`);
        }
    }

    return (
        <>
            {inputArray.map((item, index) => (
                <div key={index}>
                    {item}
                </div>
            ))}
        </>
    );
}

export default DosingMomentsInputs;