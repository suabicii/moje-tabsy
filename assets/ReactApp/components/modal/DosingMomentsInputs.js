import React from "react";
import DosingMomentInput from "./DosingMomentInput";

function DosingMomentsInputs(props) {
    const inputArray = [];

    for (const [key, value] of props.dosingMoments) {
        inputArray.push(<DosingMomentInput name={`hour${key}`} value={value}/>);
    }

    return (
        <>
            {inputArray.map((item, index) => (
                <div key={index}>
                    {item}
                </div>
            ))}
        </>
    );
}

export default DosingMomentsInputs;