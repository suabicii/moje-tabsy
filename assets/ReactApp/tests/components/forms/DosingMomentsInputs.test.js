/**
 * @jest-environment jsdom
 * */
import React from "react";
import {render, screen} from "@testing-library/react";
import DosingMomentsInputs from "../../../components/forms/DosingMomentsInputs";
import ReactShallowRenderer from "react-test-renderer/shallow";

const dosingMoments = [
    ['1', '00:00'],
    ['2', '00:01']
];

it('should correctly render dosing moments inputs', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<DosingMomentsInputs dosingMoments={dosingMoments}/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should render DosingMomentsInputs component with collection of DosingMomentInput and correct values', () => {
    render(<DosingMomentsInputs dosingMoments={dosingMoments}/>);
    expect(screen.getByTestId('hour1-test')).toHaveValue('00:00');
    expect(screen.getByTestId('hour2-test')).toHaveValue('00:01');
});