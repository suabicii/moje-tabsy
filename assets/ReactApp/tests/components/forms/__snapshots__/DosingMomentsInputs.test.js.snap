// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render dosing moments inputs 1`] = `
<React.Fragment>
  <div>
    <DosingMomentInput
      name="hour1"
      value="00:00"
    />
  </div>
  <div>
    <DosingMomentInput
      name="hour2"
      value="00:01"
    />
  </div>
</React.Fragment>
`;
